commands/dht_daemon_test.go

```diff
@@ -28,3 +31,56 @@ func TestDhtFindPeer(t *testing.T) {
 
 	assert.Contains(d2Addr, findpeerOutput)
 }
+
+func TestDhtFindProvs(t *testing.T) {
+	t.Parallel()
+	assert := ast.New(t)
+
+	d1 := makeTestDaemonWithMinerAndStart(t)
+	defer d1.ShutdownSuccess()
+
+	d2 := th.NewDaemon(t,
+		th.WithMiner(fixtures.TestMiners[0]),
+		th.KeyFile(fixtures.KeyFilePaths()[0]),
+		th.KeyFile(fixtures.KeyFilePaths()[1])).Start()
+	defer d2.ShutdownSuccess()
+
+	d3 := th.NewDaemon(t,
+		th.WithMiner(fixtures.TestMiners[0]),
+		th.KeyFile(fixtures.KeyFilePaths()[0]),
+		th.KeyFile(fixtures.KeyFilePaths()[2])).Start()
+	defer d3.ShutdownSuccess()
+
+	d1.ConnectSuccess(d2)
+	d1.ConnectSuccess(d3)
+	d2.ConnectSuccess(d3)
+
+	d1.MineAndPropagate(5*time.Second, d2)
+	d2.MineAndPropagate(5*time.Second, d1)
+	d3.MineAndPropagate(5*time.Second, d1, d2)
+
+	op1 := d1.RunSuccess("chain", "ls")
+	results := strings.Split(op1.ReadStdout(), "\n")
+	assert.True(len(results) > 1)
+	genesisKey := results[0]
+
+	t.Run("basic command succeeds", func(t *testing.T) {
+		op1 = d1.Run("dht", "findprovs", genesisKey)
+		results = strings.Split(op1.ReadStdoutTrimNewlines(), "\n")
+		// all miners should have the genesis block key
```

Comment:
BLOCKING: Whaa?  Since when are we using the dht for content routing?  I don't understand how this is passing.  If it is passing it seems like it should not be (cc @whyrusleeping).

Could be that I don't understand pubsub + dht interactions well enough but lets sync before merging this.

Reply:
FYI I had a test that I loved with all my heart in the initial `findprovs` PR that exercised both this command and the auto relay daemon flags.  However it was too flaky to make it in because I could not figure out how to guarantee a discovery time.  I am concerned that even if for some reason this should be working this test will suffer from similar flakiness.

Reply:
To be more clear: the only thing that I think this command should resolve is the cid representing the autorelay service.

Reply:
Ok I filed an issue here #2357.  This was a good bug to find.  IMO the right way to move forward with this PR is to delete all `findprovs` tests.  If you can figure out a way to test `findprovs` on the autorelay records that is not flaky then I would welcome that test but I believe that is too much to ask / out of scope for this PR.

Reply:
@ZenGround0 these tests haven't flaked on me locally; they're pretty reliable so far.